angular/projects/admin-nrpti/src/app/records/orders/order-lng-detail/order-lng-detail.component.ts

```diff
@@ -0,0 +1,96 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { takeUntil } from 'rxjs/operators';
+import { OrderLNG } from '../../../../../../common/src/app/models/lng/order-lng';
+import { Subject } from 'rxjs';
+import { ActivatedRoute, Router } from '@angular/router';
+import { RecordComponent } from '../../utils/record-component';
+import { DatePipe } from '@angular/common';
+import { FactoryService } from '../../../services/factory.service';
+
+@Component({
+  selector: 'app-order-lng-detail',
+  templateUrl: './order-lng-detail.component.html',
+  styleUrls: ['./order-lng-detail.component.scss']
+})
+export class OrderLNGDetailComponent extends RecordComponent implements OnInit, OnDestroy {
+  private ngUnsubscribe: Subject<boolean> = new Subject<boolean>();
+
+  public isLoading = true;
+  public isPublishing = false;
+  public isUnPublishing = false;
+
+  constructor(
+    public route: ActivatedRoute,
+    public router: Router,
+    public factoryService: FactoryService,
+    public datePipe: DatePipe
+  ) {
+    super();
+  }
+
+  ngOnInit() {
+    if (this.data) {
+      this.isLoading = false;
+      return;
+    }
+
+    this.route.data.pipe(takeUntil(this.ngUnsubscribe)).subscribe((res: any) => {
+      if (!res || !res.records) {
+        alert("Uh-oh, couldn't load Order");
+        this.router.navigate(['/']);
+        return;
+      }
+
+      const records = res.records[0] && res.records[0].data && res.records[0].data.searchResults;
+
+      this.data = records && records[0] && new OrderLNG(records[0]);
+
+      this.isLoading = false;
+    });
+  }
+
+  publish(): void {
+    this.isPublishing = true;
+
+    this.factoryService
+      .publishRecord(this.data)
+      .pipe(takeUntil(this.ngUnsubscribe))
+      .subscribe(response => {
+        this.data = new OrderLNG(response);
+
```

Comment:
If you're not reloading the page, you should use changeDetection incase there are other fields that have updated since originally fetching.  We should account for error cases (already published).

Same goes for all other areas in the diff.

Reply:
Added better error handling for thrown errors, and already published/unpublished cases.